src/OrchardCore.Modules/OrchardCore.UrlRewriting/Controllers/AdminController.cs

```diff
@@ -0,0 +1,200 @@
+
+        return View(shape);
+    }
+
+    [HttpPost, ActionName(nameof(CreateRule))]
+    public async Task<ActionResult> CreateRulePOST()
+    {
+        if (!await _authorizationService.AuthorizeAsync(User, UrlRewritingPermissions.ManageUrlRewriting))
+        {
+            return Forbid();
+        }
+
+        var rule = new RewriteRule();
+
+        var shape = await _rewriteRuleDisplayManager.UpdateEditorAsync(rule, updater: _updateModelAccessor.ModelUpdater, isNew: true);
+
+        if (!ModelState.IsValid)
+        {
+            return View(shape);
+        }
+
+        await _rewriteRulesStore.SaveAsync(rule);
+
+        await _notifier.SuccessAsync(H["Rule created successfully."]);
+
+        return RedirectToAction(nameof(Index));
+    }
+
+    public async Task<ActionResult> EditRule(string id)
+    {
+        if (!await _authorizationService.AuthorizeAsync(User, UrlRewritingPermissions.ManageUrlRewriting))
+        {
+            return Forbid();
+        }
+
+        var currentRule = await _rewriteRulesStore.FindByIdAsync(id);
+
+        if (currentRule == null)
+        {
+            return NotFound();
+        }
+
+        var shape = await _rewriteRuleDisplayManager.BuildEditorAsync(currentRule, updater: _updateModelAccessor.ModelUpdater, isNew: false);
+
+        return View(shape);
+    }
+
+    [HttpPost, ActionName(nameof(EditRule))]
+    public async Task<ActionResult> EditRulePOST(string id)
+    {
+        if (!await _authorizationService.AuthorizeAsync(User, UrlRewritingPermissions.ManageUrlRewriting))
+        {
+            return Forbid();
+        }
+
+        var currentRule = await _rewriteRulesStore.FindByIdAsync(id);
+
+        if (currentRule == null)
+        {
+            return NotFound();
+        }
+
+        var shape = await _rewriteRuleDisplayManager.UpdateEditorAsync(currentRule, updater: _updateModelAccessor.ModelUpdater, isNew: false);
+
+        if (ModelState.IsValid)
+        {
+            await _rewriteRulesStore.SaveAsync(currentRule);
+
+            await _notifier.SuccessAsync(H["Rule updated successfully."]);
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        return View(shape);
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> DeleteRule(string id)
+    {
+        if (!await _authorizationService.AuthorizeAsync(User, UrlRewritingPermissions.ManageUrlRewriting))
+        {
+            return Forbid();
+        }
+
+        var currentRule = await _rewriteRulesStore.FindByIdAsync(id);
+
+        if (currentRule == null)
+        {
+            return NotFound();
+        }
+
+        await _rewriteRulesStore.DeleteAsync(currentRule);
+
+        await _notifier.SuccessAsync(H["Rule deleted successfully."]);
+
+        return RedirectToAction(nameof(Index));
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> ReloadWebsite()
```

Comment:
I would rather just reload the shell every time a new rule is added or a rule edited with change instead of adding an action to reload the site. This is how to do it in other places like recipes and features. We typically add a warning from the display driver about the release.